### PR TITLE
UI Enhancements

### PR DESCRIPTION
**Pull request by jas594**
1. Disable all region checkboxes on startup
2. Grey out all region checkboxes when enabled and re-enable them when disabled

### CODE REVIEW

**Comment by jas594** [comment]
also, are there plans to merge the recent updates to the PortSwigger repo?

**Comment by DaveYesland** [comment]
I think disabling the check boxes when the gateways are deployed is a good idea.

The check boxes are intentionally all checked on startup because with the latest update now the regions which are invalid will automatically be deselected and only valid regions will be spun up anyways making it somewhat automated to use all valid regions in your account by default.

If you make a commit to this and remove the disabling check boxes on startup I can test this out and merge it.

File: IPRotate.py [hunk 1]
```diff
@@ -542,7 +556,7 @@ def getUiComponent(self):
 		for region in AVAIL_REGIONS:
 			cur_region = region.replace('-','_')
 			cur_region = cur_region+'_status'
-			setattr(self,cur_region,JCheckBox(region,True))
+			setattr(self,cur_region,JCheckBox(region,False))
```
Comment (DaveYesland) [inline IPRotate.py:559]: Change back to True since region selection is automated now if you just leave them all checked.

**Comment by DaveYesland** [comment]
> also, are there plans to merge the recent updates to the PortSwigger repo?

Yeah I will be requesting they update their repo soon.

**Comment by DaveYesland** [comment]
Closing this as there have been quite a few changes which includes different ways of handling selected and deselecting of regions.

**Pull request closed without merging**